docs/src/internals/montecarlo.md

```diff
@@ -0,0 +1,136 @@
+# Monte Carlo Simulation support
+
+## @defmcs macro
```

Comment:
I think this is nice! I also think we should probably start with the types and functions, and then the macro can be a nice wrapper/shortcut for creating those. But the text here is great.